drawpixels/src/drawpixels.cpp

```diff
@@ -62,9 +62,33 @@ static void fill_line(int fromx, int tox, int y, int r, int g, int b, int a){
     fromx = tox;
     tox = temp;
   }
-  for (int x = fromx; x <= tox; x++) {
-    putpixel(x, y, r, g, b, a);
+  fromx = max(0, fromx);
+  tox = min(buffer_info.width-1, tox);
```

Comment:
![2018-07-17_09-48-29](https://user-images.githubusercontent.com/2209596/42801000-92c617fc-89a6-11e8-8100-b62120faace6.jpg)
What abot fmax/fmin, or maybe our own defines? https://stackoverflow.com/questions/7035023/stdmax-expected-an-identifier?answertab=votes#tab-top

Reply:
Change to fmax.